.dockerignore

```diff
@@ -0,0 +1 @@
+**/.git
```

Comment:
While nice this is actually not needed as `git` seems smart enough not to initialize it in submodules. We could keep it as a safeguard though.

The real heavy weight is in `vendor` directory under `ovn-kubernetes/go-controller`. We could ignore that directory and init it when its needed using `go mod vendor`. This shaves off ~54M off the added size (~57M).

Reply:
Currently we're not doing any go builds on kubernetes code, just using the scripts. Will add, good to have it.